vyper/builtins/functions.py

```diff
@@ -1219,6 +1221,18 @@ def build_IR(self, expr, args, kwargs, contact):
         )
 
 
+class BlobHash(BuiltinFunctionT):
+    _id = "blobhash"
+    _inputs = [("index", UINT256_T)]
+    _return_type = BYTES32_T
+
+    @process_inputs
+    def build_IR(self, expr, args, kwargs, contact):
+        if not version_check(begin="cancun"):
```

Comment:
should this check move to the frontend? like if somebody requests `blobhash` but only requests `annotated_ast` should this fail? or we only fail if ir/asm/bytecode is requested

Reply:
Personally am not too opinionated about it, but I think failure if ir/asm/bytecode is requested seems fine. Is there maybe another similar case we can look at?

Reply:
hmm mostly we reject versioned features during codegen, although `transient` is rejected in the frontend

Reply:
the new block property `block.blobbasefee` is currently also planned to reject at codegen time: https://github.com/vyperlang/vyper/pull/3945/files#diff-452732981c7a1a928e9f249c7b14c723695de3d6635568ccc8899b6cab2fc54bR292-R294. We should keep this consistent at least.